platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/config/MerkleDbConfig.java

```diff
@@ -107,7 +107,9 @@ public record MerkleDbConfig(
         @ConfigProperty(defaultValue = "-1") int numHalfDiskHashMapFlushThreads,
         @ConfigProperty(defaultValue = "262144") int reservedBufferLengthForLeafList,
         @ConfigProperty(defaultValue = "1048576") int leafRecordCacheSize,
-        @ConfigProperty(defaultValue = "true") boolean usePbj) {
+        @ConfigProperty(defaultValue = "true") boolean usePbj,
+        @Min(1) @ConfigProperty(defaultValue = "8") int maxFileChannelsPerFileReader,
+        @Min(1) @ConfigProperty(defaultValue = "8") int maxThreadsPerFileChannel) {
```

Comment:
Minor: it looks like the hard-coded default values try to make assumptions about the hardware running this code. I'd suggest to consider deriving the default values from Runtime.availableProcessors() and similar so as to try and optimally use all the hardware available, and prevent overuse of the resources when the code is running on a low-end hardware, too. This may be done separately though.

Reply:
In case of the state validator there was a clear goal - reduce or eliminate the bottleneck that prevents reading values as much as possible. In this particular case it's not clear how exactly any of these values should be derived from `Runtime.availableProcessors()`. It looks like it should be measured on a particular configuration and adjusted to it. There are too many performance variables to consider here to predict what values the parameters should have.

Reply:
Fair enough.